src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -666,6 +665,11 @@ private void GeneratePropMetadataInitFunc(SourceWriter writer, string propInitMe
                         writer.WriteLine($"properties[{i}].ObjectCreationHandling = {GetObjectCreationHandlingAsStr(property.ObjectCreationHandling.Value)};");
                     }
 
+                    if (property.Order != 0)
```

Comment:
For the new naming conflict resolution to work properly, custom ordering of properties need to be run _after_ conflict resolution. Hence, for the case of the metadata source generator properties need to be sorted at run time as opposed to compile time.